foreign_cc/private/configure_script.bzl

```diff
@@ -173,7 +173,7 @@ _CONFIGURE_TOOLS = {
     "AR": "cxx_linker_static",
     "CC": "cc",
     "CXX": "cxx",
-    # missing: cxx_linker_executable
+    "LD": "cxx_linker_executable",
```

Comment:
Do you know why this was "missing" before? If the implementation is this simple 😅?

Reply:
I'm not sure either, although I think this change is the culprit behind the Linux build failures in BuildKite :(

Reply:
Please see new commit I have added to this PR regarding the linker executable

Reply:
The commit is kinda hard to read, I'm wondering if the line endings are being changed for the file since all lines have mods but it doesn't look like they have changed.

Reply:
I just realised that I don't actually need this change for OpenSSL to build. I initially made the change as i was initially building OpenSSL with no configure options. Now that OpenSSL is building with "no-shared", the LD executable is no longer required.

This change seemed to open a can of worms, which i would rather not resolve with this PR as it is a wider issue. The upshot is that it appears rules_foreign_cc cannot build shared libraries.